lib/overcommit/hook/pre_commit/go_vet.rb

```diff
@@ -0,0 +1,20 @@
+module Overcommit::Hook::PreCommit
+  # Runs `go vet` against any modified Golang files.
+  class GoVet < Base
+    def run
+      result = execute(command + applicable_files)
+      return :pass if result.success?
+
+      if result.stderr =~ /no such tool "vet"/
+        return :warn, "`go tool vet` is not installed#{install_command_prompt}"
```

Comment:
We've recently changed the preferred failure mode for dependencies to actually `:fail` instead of `:warn`. If you have the hook enabled, it should fail unless it actually ran successfully.